kolibri/plugins/coach/assets/src/composables/useResourceSelection.js

```diff
@@ -73,29 +70,6 @@ export default function useResourceSelection({
   const selectedResources = ref([]);
   const topic = ref(null);
 
-  const fetchBookmarks = async params => {
-    const response = await ContentNodeResource.fetchBookmarks(params);
-    if (bookmarks?.annotator) {
-      const annotatedResults = await bookmarks.annotator(response.results);
-      return {
-        ...response,
-        results: annotatedResults,
-        count: annotatedResults.length,
-      };
-    }
-    return response;
-  };
-  const bookmarksFetch = useFetch({
-    fetchMethod: () =>
-      fetchBookmarks({
-        params: { limit: 25, available: true, ...bookmarks?.filters },
-      }),
-    fetchMoreMethod: more =>
-      ContentNodeResource.fetchBookmarks({
-        params: more,
-      }),
-  });
-
```

Comment:
The changes in this PR make it so we no longer paginate fetching Bookmarks because of the new strategy of loading them all once and maintaining the larger list client-side.

@rtibbles do you think this would be problematic overall? The ContentNodeResource.fetchBookmarks API response includes full ContentNode data, so that might get pretty hefty if the user has lots of bookmarks.

Reply:
@AlexVelezLl are there any consequences of removing this that I may have missed? I think I got all of the places that relied on this bookmarksFetch.

Reply:
Do you think it's more problematic than the fetching all of the user's bookmarks (sans ContentNode FWIW, so smaller payload) on every navigation?

One thing I just updated is so that we still have the View More button on Select from bookmarks to avoid rendering the whole list all at once if this fetch-once-and-maintain-locally approach ends up merging.

Reply:
The current API call seems significantly less likely to timeout, and if that is the concern, why don't we just cache the bookmarks call in the current code? For some reason we are completely bypassing the built in cache mechanisms in the APIResource layer by calling `force: true` when we fetch the bookmarks, and then not allowing it to handle cache invalidation by making direct calls to the endpoint when modifying the bookmarks.

It is possible that cache invalidation may not be working completely as expected for deletion in the APIResource layer, but it seems like it would be simpler just to fix that than this refactor.

The APIResource layer was specifically setup to allow us to do these repeated API calls on page navigation but returning a cached response if desired.